src/Subscription/SubscriptionHandler.php

```diff
@@ -3,23 +3,52 @@
 namespace AppBundle\Subscription;
 
 use AppBundle\Entity\Adherent;
+use AppBundle\Entity\SubscriptionType;
+use AppBundle\Repository\AdherentRepository;
 use AppBundle\Repository\SubscriptionTypeRepository;
 
 class SubscriptionHandler
 {
-    private $repository;
+    public const ACTION_TYPE_SUBSCRIBE = 'subscribe';
+    public const ACTION_TYPE_UNSUBSCRIBE = 'unsubscribe';
 
-    public function __construct(SubscriptionTypeRepository $repository)
+    private $adherentRepository;
+    private $subscriptionTypeRepository;
+
+    public function __construct(AdherentRepository $adherentRepository, SubscriptionTypeRepository $subscriptionTypeRepository)
     {
-        $this->repository = $repository;
+        $this->adherentRepository = $adherentRepository;
+        $this->subscriptionTypeRepository = $subscriptionTypeRepository;
+    }
+
+    public function changeSubscription(string $type, string $email, string $listId): void
```

Comment:
If type is not a vlid constant please throw an `\InvalidArgumentException`. And add a test case for it?

Reply:
Done.